Tests/TaskTests.swift

```diff
@@ -165,6 +165,89 @@ class TaskTests: XCTestCase {
         waitForExpectationsWithTimeout(TestTimeout, handler: nil)
     }
 
+    func testThatCancellationIsAppliedImmediatelyWhenMapping() {
+        let beforeExpectation = expectationWithDescription("original task cancelled")
+        let beforeTask = Task<Int>(Deferred<TaskResult<Int>>(), cancellation: beforeExpectation.fulfill)
+
+        beforeTask.cancel()
+        XCTAssert(beforeTask.progress.cancelled)
+
+        let afterTask: Task<String> = beforeTask.map(impossible)
+
+        XCTAssert(afterTask.progress.cancelled)
+
+        waitForExpectationsWithTimeout(TestTimeout, handler: nil)
+    }
+
+    func testThatTaskCreatedWithProgressReflectsThatProgress() {
+        let progress = NSProgress(parent: nil, userInfo: nil)
+        progress.totalUnitCount = 10
+        progress.setUserInfoObject(true, forKey: "Test")
+        progress.cancellable = false
+
+        let task = Task<Int>(Deferred<TaskResult<Int>>(), progress: progress)
+
+        XCTAssertEqualWithAccuracy(task.progress.fractionCompleted, 0, accuracy: 0.001)
+        XCTAssertEqual(progress.userInfo["Test"] as? Bool, true)
+        XCTAssert(task.progress.cancellable)
+
+        progress.completedUnitCount = 5
+        XCTAssertEqualWithAccuracy(task.progress.fractionCompleted, 0.5, accuracy: 0.001)
+    }
+
+    func testThatTaskCreatedUnfilledIsIndeterminate() {
+        let task = Task<Int>()
+
+        XCTAssert(task.progress.indeterminate)
+    }
+
+    func testThatTaskWrappingUnfilledIsIndeterminate() {
+        let d = Deferred<Task<Int>.Result>()
+        let task = Task(d, cancellation: {})
+
+        XCTAssertFalse(task.progress.indeterminate)
+    }
+
+    func testThatTaskWrappingFilledIsDeterminate() {
+        let d = Deferred<Task<Int>.Result>.init(value: .Success(42))
+        let task = Task(d, cancellation: {})
+
+        XCTAssertFalse(task.progress.indeterminate)
+    }
+
+    func testThatMapExtendsParentFractions() {
```

Comment:
That's fair. Will add more prose around this throughout the codebase.

30 sec summary (as I'm on my phone) is that NSProgress is a tree, and Task mapping is linear. Each new transform adds 1 unit to the denominator of the "root" Task.